sourcegraph/learn
# Add copy atom & functionality for codeblocks

### What should this PR do?
Resolves [DEVED-228](https://sourcegraph.atlassian.net/browse/DEVED-228) by adding a `CopyToClipboard` atom to be used with our code blocks. When clicked, this component enables code copying via the [Clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API).

Also resolves [DEVED-244](https://sourcegraph.atlassian.net/browse/DEVED-244) by adding a separate component, `OutputHighlighter`, to enable highlighting on output blocks (bundling this in here since it seems to roll these together than to update all of the output blocks in our tutorials **again** in a separate PR).

### Why are we making this change?
- We would like users to have a delightful experience learning on our site, and being able to copy code for learning purposes will aid this goal.
- We would like to emphasize elements in our output blocks, as we are able to do in our language highlighted blocks.

### What are the acceptance criteria? 
- Code in codeblocks should be copied when the copy icon is clicked.
- The icon should have a `hover` color change.
- Code output should not have a copy button (all output blocks on all tutorials should be updated as part of this PR).
- Any `matcher` props passed to the `OutputHighlighter` component should display in yellow, as a `<mark>`.

### How should this PR be tested?
- Check out the branch, and ensure that the above criteria are met in a given tutorial.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
